scripts/prettify-staged.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
+
+prettify() {
+    FILES=$1
+    echo "Formatting staged and uncommitted files:"
+    ./node_modules/.bin/prettier --write $FILES
```

Comment:
> staged and uncommitted

Does this include files that are not staged (or only partially staged)? If yes, I would be annoyed if I unsuspectingly ran a script and it modified files that had unstaged changes since it could be difficult to untangle hand-crafted changes from automated prettifier changes.

Reply:
Yeah, and as a reviewer I much prefer when formatting changes are separate from logical changes. One way to mitigate this is to change the instructions to encourage committing the formatting separately. Another is to reformat the entire repo once this is merged (which I think we should do anyway, it'll take forever to reformat opportunistically).

Reply:
In local testing this correctly excludes files that are _not_ staged, but it does include files that are partially staged and formats them as a whole. If that's a major issue, we could use something like [git-format-staged](https://github.com/hallettj/git-format-staged) to get formatting for only those portions of files that are staged.

Reply:
@orangejenny would preferable behavior in your mind be something more like this?
- Identify which files want reformatting in pre-commit, but don't block the commit for it
- Recommend running the command to reformat those whole files
- Use the prettify-staged (or a renamed/modified version) script to optionally commit those files as well, similar to the `build_bootstrap5_diffs` command

Reply:
`git-format-staged` seems nice for its purpose (I'm also a devotee of `-p` when staging changes). I think the downside is that we'll then have files that have old formatting interspersed with new, but that's fixable by then formatting the entire file in a new commit that reviewers can largely ignore.

Reply:
(commented before seeing your comment addressed to me, and seeing your previous comment gave me more to think about) I think blocking the commit is fine, since `--no-verify` exists. It's good to have an attention-grabbing moment, and blocking the commit does that. I was basically asking to append "in a separate commit" to "Run ./scripts/prettify-staged.sh to fix them automatically."

Reply:
@orangejenny That seems reasonable. I think there would also need to be a corresponding change to the behavior of `prettify-staged` for usability. If we're asking a developer to make style fixes in a separate commit, perhaps that script should target all files that have been changed and _are_ committed relative to `master`, and maybe call it `prettify-changed` or something?